### ISSUE TITLE
Telegram window doesn't receive focus after clicking on notification popup or tray icon under Gnome and Wayland

### ISSUE THREAD

**Issue by vpkopylov** [telegramdesktop/tdesktop#8800]
### Steps to reproduce
1. Check "Launch Telegram when system starts" setting
2. Login in Gnome using Wayland (default) 
3. Open Telegram window
4. Don't close Telegram window and open any other window on top of it (for example a browser)
5. Wait for a notification popup from Telegram and click on it or click on the tray icon (I use TopIcons plus Gnome extension)

### Expected behaviour
Telegram window receives focus

### Actual behaviour
Telegram window remains in the background, although I see Telegram displayed in the Gnome's top panel. If I switch to another app again and then to Telegram it does receive focus. 

### Configuration
Arch Linux with all updated installed.
 "Launch Telegram when system starts" setting is checked, if I start Telegram manually this issue doesn't reproduce. Also the problem doesn't seem to happen under X.Org. I didn't override any env vars, so I guess Telegram is running in XWayland mode. 

**Version of Telegram Desktop:** 2.4.3

**Installation source (Linux Only):** distribution package, also tried flatpak and telegram-desktop-bin AUR package (which is a wrapper for the official binary Linux package, I guess)

**Used theme**: Default

**Comment by ilya-fedin** [comment]
There are no API to focus a window on Wayland, this is impossible to fix.

**Comment by vpkopylov** [comment]
> There are no API to focus a window on Wayland, this is impossible to fix.

Why then Telegram behaves as expected in the same Wayland session when the app started manually (not auto-started)? In that case ff Telegram window is in the background and I receive a notification while doing something in the browser when I click on this notification Telegram window definitely becomes foreground.  If Telegram is started automatically it behaves as described in the issue.

**Comment by ilya-fedin** [comment]
I guess because you have forced xwayland in regular .desktop file, while haven't forced in autostart .desktop file.

**Comment by vpkopylov** [comment]
> I guess because you have forced xwayland in regular .desktop file, while haven't forced in autostart .desktop file.

I can prove that Telegram runs in XWayland mode when auto-started by [running xwininfo command](https://fedoraproject.org/wiki/How_to_debug_Wayland_problems#Does_your_application_run_on_Wayland_natively.2C_or_uses_XWayland_.28X11_compatibility_layer.29.3F), selecting Telegram window and getting information from X.Org (Window id, etc), it is impossible if an app runs under Wayland.

**Comment by ilya-fedin** [comment]
As I already said, there is nothing could be done, except of hiding these buttons on Wayland.

**Comment by ilya-fedin** [comment]
Duplicate on #8074

**Comment by vpkopylov** [comment]
It's not a duplicate of #8074 in my opinion. 

In #8074 described the problem with pure Wayland

> Telegram is run on Wayland (therefore not with Xwayland)

But in my case the app is running under XWayland, as I explained in this [comment](https://github.com/telegramdesktop/tdesktop/issues/8800#issuecomment-706670050)

Inability to pass the focus to the app is not the case, I guess. Telegram is the current app in Gnome: I can see it in the top panel. And I even can type something and see it printed in Telegram's search bar (although Telegram window is overlapped by another app). Take a look at [this screenshot](https://i.imgur.com/RlrT30z.png). Maybe I named the issue incorrectly and it should be something like "Telegram window is not repainted after receiving focus under XWayland"

**Comment by ilya-fedin** [comment]
Can you bisect through old version and tell on which version exactly the issue appeared?

**Comment by vpkopylov** [comment]
> Can you bisect through old version and tell on which version exactly the issue appeared?

I'll try. Thanks for the assistance!

**Comment by vpkopylov** [comment]
I can reproduce the problem even with very old Telegram versions (1.5 for example). Maybe some change in Gnome caused it and not in Telegram itself.  
I compared the output of xprop for auto- and manually started Telegram and the only significant difference I found: _NET_WM_USER_TIME prop is set only if Telegram is auto-started. Maybe it somehow related.
Also, I found the following workaround, if Exec line in ~/.config/autostart/telegram.desktop is changed from `telegram-desktop -autostart` to `sh -c "sleep 1 && telegram-desktop -autostart" ` Telegram behaves as expected.
